lib/tic_tac_toe.rb

```diff
@@ -0,0 +1,167 @@
+# frozen_string_literal: true
+
+class Player
+  # identity ['X','O']
+  # name ['Player N°1', 'Player N°2']
+  # turn [true, false]
+  attr_accessor :identity, :name, :turn
+  def initialize(identity, name, turn)
+    @identity = identity
+    @name = name
+    @turn = turn
+  end
+end
+
+class Board
+  # main_board: Grid: 3rows, 3columns
+  attr_accessor :main_board
+
+  def initialize
+    @main_board = [%w[1 2 3], %w[4 5 6], %w[7 8 9]]
+  end
+
+  # return true or false
+  # 1st
+  # [['1','2','3'],['4','5','6'],['7','8','9']] => ['1','2','3','4','5','6','7','8','9']
+  # 2nd
+  # compare if in the array exist 'X' or 'O'
+  def have_space
+    @main_board.flatten.any? { |mark| mark != 'X' && mark != 'O' }
+  end
+
+  # this function save 'X' or 'O' in the number selected of the Grid
+  # e.g. if Player N°1 ('X') selected the number 2 in the first turn:
+  # | - | - | - |
+  # | 1 | X | 3 |
+  # | - | - | - |
+  # | 4 | 5 | 6 |
+  # | - | - | - |
+  # | 7 | 8 | 9 |
+  # | - | - | - |
+  # if the position in the Grid is free (without 'X' or 'O') return true,
+  # othercase it returns false
+  def check_choise_player(player, number)
```

Comment:
its a more then 60 lines method so its a high time to refactor it.
The code should respect the single responsibility rule for methods (i.e. each method does exactly one thing). Studentes tends to create long methods that are doing multiple steps of the game process.